src/plugins/mer/mergeneraloptionswidget.cpp

```diff
@@ -86,6 +90,45 @@ MerGeneralOptionsWidget::MerGeneralOptionsWidget(QWidget *parent)
     m_ui->askBeforeStartingVmCheckBox->setToolTip(m_ui->askBeforeStartingVmCheckBox->toolTip().arg(Sfdk::Sdk::osVariant()));
     m_ui->askBeforeClosingVmCheckBox->setToolTip(m_ui->askBeforeClosingVmCheckBox->toolTip().arg(Sfdk::Sdk::osVariant()));
     m_ui->rpmValidationInfoLabel->setText(m_ui->rpmValidationInfoLabel->text().arg(Sfdk::Sdk::osVariant()));
+
+    auto setSignElementsEnabled = [](Ui::MerGeneralOptionsWidget *ui, bool enabled) {
+        ui->defaultSigningUserComboBox->setEnabled(enabled);
+        ui->defaultSigningPassphraseFile->setEnabled(enabled);
+    };
+
+    connect(m_ui->signPackagesByDefaultCheckBox, &QCheckBox::toggled, this, [&](bool checked) {
+        setSignElementsEnabled(m_ui, checked);
+    });
+    m_ui->signPackagesByDefaultCheckBox->setChecked(MerSettings::signPackagesByDefault());
+
+    const bool isGpgAvailable = Sfdk::isGpgAvailable();
+    if (!isGpgAvailable) {
+        m_ui->signErrorLabel->setText(tr("GPG utility not found!"));
+        m_ui->signErrorLabel->setStyleSheet(QLatin1String("color: \"yellow\""));
```

Comment:
hm, I don't see any "yellow" usage in src/plugins. Maybe "red"? More occurrences.

Reply:
"red" looks like a fatal error. Without gpg, we can safely continue using SDK. Maybe should hide items if gpg is not available?

Reply:
Red indeed looks like a fatal error. Hiding is not good either, as the user should be given a hint that it might be a good idea to install gpg. Perhaps something like Theme::IconWarningColor.name(QColor::HexRgb) or something like that could be used?

Reply:
Without GPG, we can safely continue using SDK but we cannot use GPG and that's what this part of UI is about :)

Red highlight is what upstream QtC uses consistently. Check e.g. Options > Version Control - try to enter invalid path in any field.

Alternatively, we could also show the label using default color and show the warning icon in front of it.

And we should also think about adding path chooser here, which already provides the desired feedback, but I would prefer to postpone that as we should ideally merge this before the QtC upgrade work is started.

Reply:
> Alternatively, we could also show the label using default color and show the warning icon in front of it.

Indeed. Now I feel a bit silly as when I looked at what upstream QtC uses in such situations, I saw the warning icon. That's why I wrote about using IconWarningColor. I feel so stupid for not realizing that we could also use the icon!

Reply:
@etaishev it would be nice to fix this one yet